web/modules/custom/dpl_go/test/src/Unit/GoSiteTest.php

```diff
@@ -131,4 +161,76 @@ public function testIsGoNode(): void {
     $this->assertTrue($this->goSite->isGoNode($node->reveal()));
   }
 
+  /**
+   * Test that isGoNid() recognizes go nodes.
+   */
+  public function testIsGoNidGoNode(): void {
+    $this->nodeStorage->load('23')->willReturn($this->nodeProphecy('go_node'));
+
+    $this->state->get('dpl_go.node_type_cache_0', [])->willReturn([]);
+    $this->state->set('dpl_go.node_type_cache_0', ['23' => TRUE])->shouldBeCalled();
+
+    $this->assertTrue($this->goSite->isGoNid('23'));
+  }
+
+  /**
+   * Test that isGoNid() recognizes non-go nodes.
+   */
+  public function testIsGoNidNonGoNode(): void {
+    $this->nodeStorage->load('124')->willReturn($this->nodeProphecy('node'));
+
+    $this->state->get('dpl_go.node_type_cache_1', [])->willReturn([]);
+    $this->state->set('dpl_go.node_type_cache_1', ['124' => FALSE])->shouldBeCalled();
```

Comment:
Consider dropping `shouldBeCalled()`.

The scope of the test is not to check if caching works. This is handled by `testIsGoNidUsesCache()`.

Reply:
Well, as there's a prophecy on the object, it'll complain about unexpected calls. One could make a `willReturn` instead, but I think that's more confusing.